src/test/java/StringManipulationTest.java

```diff
@@ -1,149 +1,201 @@
     }
 
     @Test
     public void testRemoveNthCharacter2() {
-        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        manipulatedString.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?"); //setting string
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedString.removeNthCharacter(3, true)); //asserting removal with spaces
     }
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Example text for test case."); //setting string
+        assertEquals("Eapetx o etcs.", manipulatedString.removeNthCharacter(2, false)); //asserting removal of characters
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Example text for test case."); //setting string
+        assertEquals("Example text for test case.", manipulatedString.removeNthCharacter(1, false)); //asserting no change for n=1
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Small"); //setting string
+        assertEquals("Sal", manipulatedString.removeNthCharacter(2, false)); //asserting removal of characters
     }
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Small"); //setting string
+        assertEquals("S a l", manipulatedString.removeNthCharacter(2, true)); //asserting removal with spaces
     }
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedString.removeNthCharacter(20, true)); //asserting exception for out of bounds
     }
 
     @Test
-    public void testGeSubStrings1() {
-        manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+    public void testGetSubStrings1() {
+        manipulatedString.setString("This is my string"); //setting string
+        String [] sStings = manipulatedString.getSubStrings(3, 4); //get substring array
 
-        assertEquals(sStings[0], "my");
+        assertEquals(sStings[0], "my"); //asserting elements
         assertEquals(sStings[1], "string");
     }
 
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2() {
+        manipulatedString.setString("A B C D E"); //setting string
+        String[] subStrings = manipulatedString.getSubStrings(2, 4); //get substring array
+        assertEquals("B", subStrings[0]); //asserting elements
+        assertEquals("C", subStrings[1]);
+        assertEquals("D", subStrings[2]);
```

Comment:
Alvin, this is a really good basic test that should always be included in a method like this. I like the alteration you made from testGetSubStrings1() where you switch the two parameters in assertEquals() to test the equality both ways. I know that doesn't make a major difference in the test, but it's a minor detail that I overlooked when making my own test cases. It's all too often that the minor things we overlook in code lead to issues so I'll keep this in mind when writing my own future test cases!